src/status_im/contexts/wallet/add_account/create_account/import_private_key/view.cljs

```diff
@@ -12,45 +13,50 @@
     [utils.re-frame :as rf]))
 
 (defn- address-input
-  [{:keys [input-value set-input-value set-flow-state error?]}]
-  (let [check-address (rn/use-callback
-                       (debounce/debounce (fn [v]
-                                            (if (empty? v)
-                                              (set-flow-state nil)
-                                              ;; TODO check for validation
-                                              (if-not (v/private-key? v)
-                                                (set-flow-state :invalid-private-key)
-                                                ;; TODO add real requests
-                                                (do
-                                                  (set-flow-state :scanning)
-                                                  (js/setTimeout
-                                                   set-flow-state
-                                                   400
-                                                   (rand-nth [:active-address :inactive-address]))))))
-                                          500))
-        on-change     (rn/use-callback
-                       (fn [v]
-                         (set-input-value v)
-                         (check-address v)))
-        on-paste      (rn/use-callback
-                       (fn []
-                         (clipboard/get-string
-                          (fn [clipboard]
-                            (when-not (empty? clipboard)
-                              (on-change clipboard))))))]
+  [{:keys [input-value set-flow-state error?]}]
+  (let [check-address
+        (rn/use-callback
+         (debounce/debounce
+          (fn [v]
+            (if (empty? v)
+              (set-flow-state nil)
+              ;; TODO check for validation
+              (if-not (v/private-key? v)
+                (set-flow-state :invalid-private-key)
+                ;; TODO add real requests
+                (do
+                  (set-flow-state :scanning)
+                  ;; TODO get real address
+                  ;; Should be fixed in #18819
+                  (rf/dispatch [:wallet/set-public-address
+                                "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045"])
+                  (js/setTimeout set-flow-state 400 (rand-nth [:active-address :inactive-address]))))))
+          500))
+
+        on-change (rn/use-callback
+                   (fn [v]
+                     (rf/dispatch [:wallet/set-private-key v])
+                     (check-address v)))
+        on-paste (rn/use-callback
+                  (fn []
+                    (clipboard/get-string
+                     (fn [clipboard]
+                       (when-not (empty? clipboard)
+                         (on-change clipboard))))))]
```

Comment:
Small tweak 🔧 

We may want to update these `use-callback` usages and provide them with their dependency arrays (?). For example: `check-address` depends on `set-flow-state`, `on-change` depends on `check-address`, and `on-paste` depends on `on-change`.

Reply:
As I understand, the `set-flow-state` provided from `use-state` hook never changes

Reply:
adding use-callback in this way is questionable enough already, I think there is even further questions to adding state setters in the dependency array as I can only assume React has already considered this to some extent and as @Rende11 pointed out this method never changes so there is most likely no value in doing so 👍 
If anything we are just making the code really hard to read 🦀